astropy/utils/data.py

```diff
@@ -1899,13 +1899,14 @@ def _get_download_cache_loc(pkgname="astropy"):
         elif not os.path.isdir(datadir):
             raise OSError(f"Data cache directory {datadir} is not a directory")
```

Comment:
This can be simplified greatly using `pathlib`, but that's out of scope of this PR.